server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/IpConnectionCheck.java

```diff
@@ -0,0 +1,47 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.imapserver.netty;
+
+import java.net.InetSocketAddress;
+import java.util.Set;
+
+import org.apache.james.imap.api.ConnectionCheck;
+import org.reactivestreams.Publisher;
+
+import reactor.core.publisher.Mono;
+
+public class IpConnectionCheck implements ConnectionCheck {
+    private Set<String> bannedIps = Set.of();
+
+    @Override
+    public Publisher<Void> validate(InetSocketAddress remoteAddress) {
+        String ip = remoteAddress.getAddress().getHostAddress();
+        // check against bannedIps
+        if (bannedIps.stream().anyMatch(bannedIp -> bannedIp.equals(ip))) {
+            return Mono.error(() -> new RuntimeException("Banned"));
+        } else {
+            return Mono.empty();
+        }
+    }
+
+    public void setBannedIps(Set<String> bannedIps) {
```

Comment:
Because I need to add bannedIps for testing
```
private void addBannedIps(String clientIp) {
            imapServer.getConnectionChecks().stream()
                .filter(check -> check instanceof IpConnectionCheck)
                .map(check -> (IpConnectionCheck) check)
                .forEach(ipCheck -> ipCheck.setBannedIps(Set.of(clientIp)));
        }
```

Reply:
If it is a method for testing

1.  you can add annotation it by @VisibleForTesting
2. the `bannedIps` is always empty on prod -> the class `IpConnectionCheck` does not make sense

Reply:
ah, my mistake, it is in the "test" package. 
forget it